aesd-char-driver/main.c

```diff
@@ -162,11 +169,60 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
 	mutex_unlock(&device->lock);	
 	return retval;
 }
+
+
+
+loff_t aesd_llseek(struct file *filp, loff_t offset, int whence)
+{
+	loff_t ret_val;
+	struct aesd_dev *dev = filp->private_data;
+	loff_t size = dev->device_buffer.filledBufferSize;
+	//include lock
+	if(mutex_lock_interruptible(&dev->lock))
```

Comment:
@dhirajbennadi what are you locking here?  Is `dev` structure locking necessary?